### PR TITLE
Uncontained pipeline

### PR DESCRIPTION
**Pull request by etsauer**
#### What is this PR About?
Adds a pipeline for buidling, running, and promoting to openshift online v3 pro.

#### How should we test or review this PR?
There is a new `pipeline.md` file that explains the operations of the site. Use this to test/review the process.

#### Is there a relevant Trello card or Github issue open for this?
n/a

#### Who would you like to review this?
cc: @sabre1041 @JaredBurck @rdebeasi

### CODE REVIEW

**Comment by JaredBurck** [comment]
Getting the following error on the jenkins build at the promotion stage.
```
Exiting "Verify OpenShift Deployment" successfully; deployment "site-2" has completed with status:  [Complete].  The deployment reached "2" replica(s).
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] podTemplate
[Pipeline] {
[Pipeline] node

Running on kubernetes-927ceb4203f84cf68dd350a10eba7b2f-26a7e3da34f8 in /home/jenkins/workspace/field-guides-dev-site-pipeline
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Promote To field-guides-prod)
[Pipeline] container
[Pipeline] {

[Pipeline] sh

[field-guides-dev-site-pipeline] Running shell script
Executing shell script inside container [jenkins-slave-image-mgmt] of pod [kubernetes-927ceb4203f84cf68dd350a10eba7b2f-26a7e3da34f8]
Executing command: sh -c echo $$ > '/home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/pid'; jsc=durable-e291ca2be53e7da2ab9e287c10a4d233; JENKINS_SERVER_COOKIE=$jsc '/home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/script.sh' > '/home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/jenkins-log.txt' 2>&1; echo $? > '/home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/jenkins-result.txt' 
[?1034hsh-4.2$ cd "/home/jenkins/workspace/field-guides-dev-site-pipeline"
sh-4.2$ sh -c echo $$ > '/home/jenkins/workspace/field-guides-dev-site-pipeline@ 
tmp/durable-0f64d6f5/pid'; jsc=durable-e291ca2be53e7da2ab9e287c10a4d233; JENKINS 
_SERVER_COOKIE=$jsc '/home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/ 
durable-0f64d6f5/script.sh' > '/home/jenkins/workspace/field-guides-dev-site-pip 
eline@tmp/durable-0f64d6f5/jenkins-log.txt' 2>&1; echo $? > '/home/jenkins/works 
pace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/jenkins-result.txt' 
sh: /home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/pid: Permission denied
sh: /home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/jenkins-log.txt: Permission denied
sh: /home/jenkins/workspace/field-guides-dev-site-pipeline@tmp/durable-0f64d6f5/jenkins-result.txt: Permission denied
sh-4.2$ exit
exit

command terminated with non-zero exit code: Error executing in Docker Container: 1[Pipeline] }
[Pipeline] // container
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] End of Pipeline
ERROR: script returned exit code -2
Finished: FAILURE
```

File: pipeline.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+oc adm policy add-role-to-user edit -z promoter -n field-guides-prod
+oc process -f deploy/template.yml --param-file=deploy/prod/params | oc apply -f -
+```
+
+Now, grab the token value for the service account created above and save for later
+```
+TOKEN=$(oc serviceaccounts get-token promoter -n field-guides-prod)
+```
+
+Log back into the Development Cluster, and create the production secret, passing the Clusters API URL, Registry Hostname, and the Token from above.
```
Comment (sabre1041) [inline pipeline.md]: Need to explain how to create the `prod-credentials` secret

**Comment by sabre1041** [comment]
@JaredBurck can you verify the images in the `field-guides-dev` namespace. I have a feeling that you may be missing the `jenkins-slave-image-mgmt` image

**Comment by JaredBurck** [comment]
@sabre1041  here is my IS
```$ oc project
Using project "field-guides-dev" on server "https://192.168.64.13:8443".
$ oc describe is jenkins-slave-image-mgmt
Name:			jenkins-slave-image-mgmt
Namespace:		field-guides-dev
Created:		30 minutes ago
Labels:			role=jenkins-slave
			template=jenkins-slave-image-mgmt
Annotations:		kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"ImageStream","metadata":{"annotations":{"slave-label":"jenkins-slave-image-mgmt"},"labels":{"role":"jenkins-slave","template":"jenkins-slave-image-mgmt"},"name":"jenkins-slave-image-mgmt","namespace":"field-guides-dev"}}

			slave-label=jenkins-slave-image-mgmt
Docker Pull Spec:	172.30.1.1:5000/field-guides-dev/jenkins-slave-image-mgmt
Image Lookup:		local=false
Unique Images:		1
Tags:			1

latest
  pushed image

  * 172.30.1.1:5000/field-guides-dev/jenkins-slave-image-mgmt@sha256:f0d0e0f085ebfb6a45a666e9b44e00109b66ad5c94c17a2a4113591d0d837c6b
      26 minutes ago```

**Comment by sabre1041** [comment]
@JaredBurck Can you export the pod the slave running the build at the time of failure?

**Comment by sabre1041** [comment]
@etsauer LGTM

**Comment by JaredBurck** [comment]
@sabre1041 pod from from failing slave
```
cat kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db
apiVersion: v1
kind: Pod
metadata:
  annotations:
    openshift.io/scc: anyuid
  creationTimestamp: 2017-11-14T08:01:29Z
  deletionGracePeriodSeconds: 30
  deletionTimestamp: 2017-11-14T08:03:03Z
  labels:
    jenkins: slave
    jenkins/promotion-slave: "true"
  name: kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db
  namespace: field-guides-dev
  resourceVersion: "2503"
  selfLink: /api/v1/namespaces/field-guides-dev/pods/kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db
  uid: 05abd432-c912-11e7-a01c-5e3e6e0b391b
spec:
  containers:
  - command:
    - cat
    env:
    - name: JENKINS_LOCATION_URL
    - name: JENKINS_SECRET
      value: 8dd26aac92143f63c8c77b6a5cb3ddcd885f78e98efd968315fbcd27cdc5d2cb
    - name: JENKINS_JNLP_URL
      value: http://172.30.165.66:80/computer/kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db/slave-agent.jnlp
    - name: JENKINS_TUNNEL
      value: 172.30.144.225:50000
    - name: JENKINS_NAME
      value: kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db
    - name: JENKINS_URL
      value: http://172.30.165.66:80
    - name: HOME
      value: /home/jenkins
    image: 172.30.1.1:5000/field-guides-dev/jenkins-slave-image-mgmt
    imagePullPolicy: IfNotPresent
    name: jenkins-slave-image-mgmt
    resources: {}
    securityContext:
      capabilities:
        drop:
        - MKNOD
        - SYS_CHROOT
      privileged: false
      seLinuxOptions:
        level: s0:c8,c2
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    tty: true
    volumeMounts:
    - mountPath: /home/jenkins
      name: workspace-volume
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: jenkins-token-f3426
      readOnly: true
    workingDir: /home/jenkins
  - args:
    - 8dd26aac92143f63c8c77b6a5cb3ddcd885f78e98efd968315fbcd27cdc5d2cb
    - kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db
    env:
    - name: JENKINS_LOCATION_URL
    - name: JENKINS_SECRET
      value: 8dd26aac92143f63c8c77b6a5cb3ddcd885f78e98efd968315fbcd27cdc5d2cb
    - name: JENKINS_JNLP_URL
      value: http://172.30.165.66:80/computer/kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db/slave-agent.jnlp
    - name: JENKINS_TUNNEL
      value: 172.30.144.225:50000
    - name: JENKINS_NAME
      value: kubernetes-f7892603f3544d4592803d7eba45215f-111866e92db
    - name: JENKINS_URL
      value: http://172.30.165.66:80
    - name: HOME
      value: /home/jenkins
    image: jenkinsci/jnlp-slave:2.62-alpine
    imagePullPolicy: IfNotPresent
    name: jnlp
    resources: {}
    securityContext:
      capabilities:
        drop:
        - MKNOD
        - SYS_CHROOT
      privileged: false
      seLinuxOptions:
        level: s0:c8,c2
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /home/jenkins
      name: workspace-volume
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: jenkins-token-f3426
      readOnly: true
    workingDir: /home/jenkins
  dnsPolicy: ClusterFirst
  imagePullSecrets:
  - name: jenkins-dockercfg-0fnrl
  nodeName: localhost
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext:
    seLinuxOptions:
      level: s0:c8,c2
  serviceAccount: jenkins
  serviceAccountName: jenkins
  terminationGracePeriodSeconds: 30
  volumes:
  - emptyDir: {}
    name: workspace-volume
  - name: jenkins-token-f3426
    secret:
      defaultMode: 420
      secretName: jenkins-token-f3426
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2017-11-14T08:01:30Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2017-11-14T08:02:16Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2017-11-14T08:01:29Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://ece64d35442aa35368bf3ce4e1993e0d71b5df980806523be0c93afd20a95013
    image: 172.30.1.1:5000/field-guides-dev/jenkins-slave-image-mgmt:latest
    imageID: docker-pullable://172.30.1.1:5000/field-guides-dev/jenkins-slave-image-mgmt@sha256:0180f2b2bc33bb13ea59ffc5592472b4dee8b8c1af1a302ab723f402df410aa7
    lastState: {}
    name: jenkins-slave-image-mgmt
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2017-11-14T08:01:32Z
  - containerID: docker://f262fe7c58460323ae5dc99a38c2280857cdc75e6cacbe7d836a7b0b58c1b736
    image: docker.io/jenkinsci/jnlp-slave:2.62-alpine
    imageID: docker-pullable://docker.io/jenkinsci/jnlp-slave@sha256:fc17304b6e26ba2227519bf6391c034c84094ab2d00c1e35de9f2c8f77de694a
    lastState: {}
    name: jnlp
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2017-11-14T08:02:15Z
  hostIP: 192.168.64.16
  phase: Running
  podIP: 172.17.0.2
  qosClass: BestEffort
  startTime: 2017-11-14T08:01:30Z
```

**Pull request merged**